datamade/just-spaces
# PLDP Form Elements and RESTful data handler URLs

## Overview

Hello @reginafcompton. I'm trying out this new PR template from Jean! As I was waiting for us to work through an approach to the data handler in #25, I got a start on some changes that came out of our call with UCD this week. Specifically, in this PR I:

- Implemented a Study form element plugin, which lets users select an initial Study value to attach to a Survey.
- Implemented Survey Method and Survey Representation plugins, using choices from `django-pldp`. For these two and the Study element, users can change this when they run a survey. It could be an option to hide this element if survey-makers don't want survey-runners to change the presets. 
- Removed the gender and age widgets intended for intercept surveys
- Added RESTful URLs to Fobi's data handler
- Added a delete button to delete surveys. This is mostly to make development easier, so we're not overrun with test surveys!

## Testing Instructions

1. To test the new Study element, you'll need a StudyArea. We don't yet have an interface for adding those, so in your local just-spaces repo run `python manage.py shell` and then create one:

```python
>>> from pldp.models import StudyArea
>>> polygon = 'POLYGON((-101.744384 39.32155, -101.552124 39.330048, -101.403808 39.330048, -101.332397 39.364032, -101.744384 39.32155))'
>>> StudyArea.objects.create(name='Test study area', area=polygon)
```

2. You should then be able to create a new survey at http://127.0.0.1:8000/studies/create/
3. THEN, navigate to a new survey and try adding a Study element to it. You should see 'Test study area' as an option in the Study form dropdown. When you save the element, the study you selected should be the default value.
